authorization/src/main/kotlin/com/ritense/authorization/permission/condition/ReflectingPermissionCondition.kt

```diff
@@ -16,12 +16,25 @@
 
 package com.ritense.authorization.permission.condition
 
+import java.lang.reflect.Field
+
 abstract class ReflectingPermissionCondition(type: PermissionConditionType) : PermissionCondition(type) {
     protected fun findEntityFieldValue(entity: Any, field: String): Any? {
         var currentEntity: Any? = entity
         val fields = field.split('.')
         fields.forEachIndexed { index, value ->
-            val declaredField = currentEntity!!.javaClass.getDeclaredField(value)
+            var classToSearch: Class<*>? = entity.javaClass
+            var declaredField: Field? = null
+
+            while (declaredField == null && classToSearch != null) {
+                declaredField = classToSearch.declaredFields.firstOrNull { it.name == value }
+                if (declaredField == null)
+                    classToSearch = classToSearch.superclass
+            }
+
+            if (declaredField == null) {
+                throw IllegalArgumentException("Field $field not found in class ${entity.javaClass}")
+            }
             declaredField.trySetAccessible()
```

Comment:
Usage of trySetAccessible is discouraged, and would not be needed if we only supported public fields (see comment above)

Reply:
It's needed as might be applying AC to classes we do not create ourselves.

Reply:
Again: if we only supported public fields this would not be needed right? I don't see why we would need to support non-public fields.